src/zarr/testing/store.py

```diff
@@ -251,3 +252,28 @@ async def test_list_dir(self, store: S) -> None:
 
         keys_observed = await _collect_aiterator(store.list_dir(root + "/"))
         assert sorted(keys_expected) == sorted(keys_observed)
+
+    async def test_with_mode(self, store: S) -> None:
+        data = b"0000"
+        self.set(store, "key", self.buffer_cls.from_bytes(data))
+        assert self.get(store, "key").to_bytes() == data
+
+        for mode in ["r", "a"]:
```

Comment:
I would like to see a test that verifies that a store rejects writes if it has been `with_mode('r')`'ed

Reply:
ditto

Reply:
Fixed.